src/mantine-core/src/components/Select/SelectRightSection/SelectRightSection.tsx

```diff
@@ -9,21 +9,36 @@ export interface SelectRightSectionProps {
   onClear?: () => void;
   size: MantineSize;
   error?: any;
+  disabled?: boolean;
 }
 
+const useStyles = createStyles(() => ({
+  button: {
+    '&:disabled': {
+      backgroundColor: 'transparent',
```

Comment:
Without this, the clear button looked off with the background and border. But what do you think?
![image](https://user-images.githubusercontent.com/9464690/155848866-b82c8cc0-0fd6-489d-88f0-e41b0ab79450.png)

Reply:
I think it should not be locked, just remove it when field is disabled

Reply:
Okay, this has been updated.

If disabled, the right side clear will be hidden.